tuna.go

```diff
@@ -879,29 +904,33 @@ func handlePaymentStream(
 	stream *smux.Stream,
 	npc *nkn.NanoPayClaimer,
 	lastClaimed *common.Fixed64,
-	getTotalCost func() common.Fixed64,
+	bytesPaid *common.Fixed64,
 	lastUpdate *time.Time,
 	isClosed *bool,
+	getTotalCost func() (common.Fixed64, common.Fixed64),
 ) error {
 	for {
 		tx, err := ReadVarBytes(stream)
 		if err != nil {
 			return fmt.Errorf("couldn't read payment stream: %v", err)
 		}
 
+		totalCost, totalBytes := getTotalCost()
+
 		err = nanoPayClaim(tx, npc, lastClaimed)
 		if err != nil {
 			log.Println("Couldn't claim nanoPay:", err)
 			continue
 		}
+		*bytesPaid = totalBytes
+		now := time.Now()
+		diff := now.Sub(*lastUpdate)
+		*lastUpdate = now
 
-		totalCost := getTotalCost()
-		*lastUpdate = time.Now()
-
-		if float64(*lastClaimed) < float64(totalCost)*MinTrafficCoverage {
+		if float64(*lastClaimed) < float64(totalCost)*MinTrafficCoverage && diff.Seconds() > 30 {
```

Comment:
This check can be bypassed as longer as user pay a tiny amount (e.g 1% of cost) every 30s